pandas/tests/extension/base/interface.py

```diff
@@ -29,6 +29,30 @@ def test_can_hold_na_valid(self, data):
         # GH-20761
         assert data._can_hold_na is True
 
+    def test_contains(self, data, data_missing):
+        # GH-37867
+        # Tests for membership checks. Membership checks for nan-likes is tricky and
+        # the settled on rule is: `nan_like in arr` is True if nan_like is
+        # arr.dtype.na_value and arr.isna().any() is True. Else the check returns False.
+
+        for this_data in [data, data_missing]:
+            scalar = this_data[~this_data.isna()][0]
+
+            assert scalar in this_data
+
+            na_value = this_data.dtype.na_value
+
+            if this_data.isna().any():
+                assert na_value in this_data
+            else:
+                assert na_value not in this_data
+
+            # this_data can never contain other nan-likes than na_value
+            for na_value_type in {None, np.nan, pd.NA, pd.NaT}:
+                if na_value_type is na_value:
+                    continue
+                assert na_value_type not in this_data
```

Comment:
Could also write it like this without a for-loop (and with ensuring we are actually testing with one array without missing values, and without the if isna().any() check), which I find easier to follow (but that's probably subjective ;))

```suggestion
        na_value = data.dtype.na_value
        # ensure data without missing values
        data = data[~data.isna()]

        # first elements are non-missing
        assert data[0] in data
        assert data_missing[0] in data_missing

        # check the presence of na_value
        assert na_value in data_missing
        assert na_value not in data

        # the data can never contain other nan-likes than na_value
        for na_value_type in {None, np.nan, pd.NA, pd.NaT}:
            if na_value_type is na_value:
                continue
            assert na_value_type not in data
            assert na_value_type not in data_missing
```

Reply:
Yeah, I like that better too.